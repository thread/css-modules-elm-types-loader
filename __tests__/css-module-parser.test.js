const {
  cssFileNameToElmFileName,
  generateElmModuleForStyles,
} = require('../src/css-module-parser')

describe('cssFileNameToElmFileName', () => {
  it('writes to the same directory with a file called Styles.elm', () => {
    const filename = cssFileNameToElmFileName('frontend/app/EmailInput/email_input.scss')
    expect(filename).toEqual('frontend/app/EmailInput/Styles.elm')
  })
})

describe('generateElmModuleForStyles', () => {
  const cssClasses = ['inputField', 'formHeading']

  const filename = 'frontend/app/EmailInput/email_input.scss'

  it('generates the right module name', () => {
    const result = generateElmModuleForStyles(cssClasses, filename)
    expect(result).toContain('module EmailInput.Styles exposing (Styles)')
  })

  it('generates a comment so the user knows not to edit by hand', () => {
    const result = generateElmModuleForStyles(cssClasses, filename)
    expect(result).toContain('-- This is auto generated by Webpack')
    expect(result).toContain('-- whenever email_input.scss is edited')
    expect(result).toContain('-- you do not want to edit this by hand')
  })

  it('generates the right type alias', () => {
    const result = generateElmModuleForStyles(cssClasses, filename)
    expect(result).toContain(`type alias Styles = {
  inputField: String, formHeading: String
}`)
  })

  it('matches the snapshot', () => {
    const result = generateElmModuleForStyles(cssClasses, filename)
    expect(result).toMatchSnapshot()
  })
})
