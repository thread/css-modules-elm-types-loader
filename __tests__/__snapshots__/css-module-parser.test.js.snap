// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateElmModuleForStyles matches the snapshot 1`] = `
"module EmailInput.Styles exposing (Styles)

-- This is auto generated by Webpack
-- whenever email_input.scss is edited
-- you do not want to edit this by hand

type alias Styles = {
  inputField: String, formHeading: String
}
  "
`;
